packages/hub/services/scheduled-payments/executor.ts

```diff
@@ -0,0 +1,157 @@
+import { inject } from '@cardstack/di';
+import { ScheduledPayment } from '@prisma/client';
+import { isBefore, subDays } from 'date-fns';
+import shortUUID from 'short-uuid';
+import { nowUtc } from '../../utils/dates';
+import config from 'config';
+import { Wallet } from 'ethers';
+import { JsonRpcProvider } from '@cardstack/cardpay-sdk';
+
+export let supportedChains = [
+  {
+    id: 1,
+    name: 'ethereum',
+    rpcUrl: 'https://eth-mainnet.public.blastapi.io',
+  },
+  {
+    id: 5,
+    name: 'goerli',
+    rpcUrl: 'https://eth-goerli.public.blastapi.io',
+  },
+  {
+    id: 100,
+    name: 'gnosis',
+    rpcUrl: 'https://rpc.gnosischain.com',
+  },
+  {
+    id: 77,
+    name: 'sokol',
+    rpcUrl: 'https://sokol.poa.network',
+  },
+  {
+    id: 137,
+    name: 'polygon',
+    rpcUrl: 'https://polygon-rpc.com',
+  },
+  {
+    id: 80001,
+    name: 'mumbai',
+    rpcUrl: 'https://polygon-testnet.public.blastapi.io',
+  },
+];
+
+export default class ScheduledPaymentsExecutorService {
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  scheduledPaymentFetcher = inject('scheduled-payment-fetcher', { as: 'scheduledPaymentFetcher' });
+  cardpay = inject('cardpay');
+  web3 = inject('web3-http', { as: 'web3' });
+  workerClient = inject('worker-client', { as: 'workerClient' });
+
+  async executePayment(scheduledPayment: ScheduledPayment) {
+    let rpcUrl = supportedChains.find((chain) => chain.id === scheduledPayment.chainId)?.rpcUrl;
+    let provider = new JsonRpcProvider(rpcUrl, scheduledPayment.chainId);
+    let signer = new Wallet(config.get('hubPrivateKey'));
```

Comment:
This is the crank's private key. We'll have to generate the private key for staging and production and adjust the infra, and configure the crank's address in the config contract. Is there an existing private key we can use?